Lib/subprocess.py

```diff
@@ -1480,7 +1480,16 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
             if shell:
                 startupinfo.dwFlags |= _winapi.STARTF_USESHOWWINDOW
                 startupinfo.wShowWindow = _winapi.SW_HIDE
-                comspec = os.environ.get("COMSPEC", "cmd.exe")
+                # gh-101283: with no full path, Windows looks into a
+                # current directory first so no plain "cmd.exe".
+                default_shell = "C:\\WINDOWS\\system32\\cmd.exe"
+                comspec = os.environ.get("COMSPEC", default_shell)
```

Comment:
This shouldn't assume the default location of the Windows directory, i.e. the base/root directory of the OS on the system drive. It can be installed elsewhere. We should probably implement `GetSystemWindowsDirectoryW()` or `GetwindowsDirectoryW()`[^1] in the `_winapi` module. Barring that, we should make a best effort to use the "SystemRoot" and "SystemDrive" environment variables if they're defined and not *empty strings*.

```suggestion
                system_drive = os.environ.get('SystemDrive') or 'C:'
                system_root = os.environ.get('SystemRoot') or os.path.join(
                    system_drive, 'Windows')
                comspec = os.environ.get('ComSpec') or os.path.join(
                    system_root, 'cmd.exe')
```

[^1]: On a terminal server (i.e. a server that supports remote desktop sessions), the path that's returned by `GetwindowsDirectoryW()` will be "%UserProfile%\\Windows" instead of "%SystemRoot%\\Windows" if the application executable lacks the flag `IMAGE_DLLCHARACTER­ISTICS_TERMINAL_SERVER_AWARE`. This should never be the case for the normal "python[w].exe" binaries. It could be the case if Python is embedded. However, it's extremely unlikely nowadays that any PE image lacks this flag. VC++ has set it by default for many years.